http-gateways/IPLD_GATEWAY.md

```diff
@@ -0,0 +1,196 @@
+
+The supported ADL names will vary based on gateway.
+
+An example value would be `;adl=hamt` to specify the [HAMT](https://ipld.io/specs/advanced-data-layouts/hamt/) ADL that's used to represent large maps.
+
+For example `/ipld/bafyreic672jz6huur4c2yekd3uycswe2xfqhjlmtmm5dorb6yoytgflova;adl=hamt/yes` (taken from the HAMT examples), should resolve to the following:
+
+```
+[
+  {
+    "line": 9,
+    "column": 501
+  }
+]
+```
+
+#### Schema (segment parameter)
+
+Segments may contain a `schema` key which points to the `CID` of an [IPLD Schema](https://ipld.io/docs/schemas/) in its [DMT](https://ipld.io/specs/schemas/#dsl-vs-dmt) form.
+If a `schema` key is provided, there must also be a `type` parameter which references one of the named Types within the IPLD Schema.
+
+The node at that point in the traversal will then be transformed by the schema, and any typed nodes it links to will also be transformed by their respective schemas.
+
+For example, given the following schema (note it is written in DSL form, but must be converted to the DMT in order to be refernced):
+
+```ipldschema
+type Example struct {
+  Hello String
+  Goodbye &NestedExample
+} representation tuple
+
+type NestedExample struct {
+  region String
+} representation tuple
+```
+
+The CID for the DMT of this schema is `bafyreibvheoym4avfsjfw63yhsymovm7o54ftcnxwxovqf5xxcbjddanze`
+
+A raw node of type `NestedExample`, whose CID is `bafyreia5mssvef4owvyols2bduwxl6csvlb35oigyj4gc7wm6wzg44udtq`:
+
+```json
+['Cyberspace']
+```
+
+A raw node of type `Example` which references the first node and whose CID is `bafyreifuyjaq3u3izc7qaf4shh76lk6565e72njgjxtava7q4s7bxheyxa`:
+
+```json
+['Hello', {'/': 'bafyreia5mssvef4owvyols2bduwxl6csvlb35oigyj4gc7wm6wzg44udtq'}]
+```
+
+We can construct the path `/ipld/bafyreifuyjaq3u3izc7qaf4shh76lk6565e72njgjxtava7q4s7bxheyxa/Goodbye?schema=bafyreibvheoym4avfsjfw63yhsymovm7o54ftcnxwxovqf5xxcbjddanze&type=Example`, or more succinctly `ipld://${cid2}/Goodbye?schema=${schemaCID}&type=Example`.
+
+The resolved node should look like:
+
+```JSON
+{
+	"region": "Cyberspace"
+}
+```
+
+#### Escaping / Encoding
+
+IPLD path segments and path segment keys/values may use [escape sequences that follow RFC1738](https://www.rfc-editor.org/rfc/rfc1738) to represent raw values like `/` which would otherwise be interpreted by URL parsers as being structurally significant.
+
+Specifically, any values in path segments that are part of the "reserved" list of characters `";" | "/" | "?" | ":" | "@" | "&" | "="`, or are non-ascii characters, must be escaped when encoding to the path.
+
+For example, path segment name `escape;this` should be escaped to `escape%3Bthis` so that `this` doesn't get accidentally parsed as a parameter.
+
+Similarly, the path segment name `😁` should be escaped to `%F0%9F%98%81`.
+
+## Interaction with URLs
+
+`/ipld/` HTTP paths map directly to `ipld://` URLs.
```

Comment:
I suggest we cover the entire URI family, so this spec is bit more future-proof:

```suggestion
`/ipld/` HTTP paths map directly to `urn:ipld:` URNs,  `ipld:` URIs or `ipld://` pseudo-URLs.
```